polaris-react/src/components/TextField/TextField.scss

```
@@ -202,8 +202,8 @@ $spinner-icon-size: 12px;
   // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
   left: 0;
   background-color: var(--p-surface);
-  border: var(--p-border-width-1) solid var(--p-border-subdued);
-  border-top-color: var(--p-border-shadow);
+  border: var(--p-border-width-1) solid var(--p-border-shadow);
+  border-top-color: var(--p-border);
```

```suggestion
  border-top-color: #898F94;
```

Suggesting to update to this to the hard coded value for now. This will ensure the border passes contrast for all backgrounds: app background, card, subdued cards.